python/tvm/driver/tvmc/workspace_pools.py

```diff
@@ -0,0 +1,199 @@
+                Example usage: --workspace-pools-clock-frequency-hz=flash:70000000""",
+    )
+    parser.add_argument(
+        "--workspace-pools-read-bandwidth-bytes-per-cycle",
+        nargs="?",
+        help="""The read bandwidth of the memory pool in bytes/cycle.
+                Example usage: --workspace-pools-read-bandwidth-bytes-per-cycle=flash:4""",
+    )
+    parser.add_argument(
+        "--workspace-pools-write-bandwidth-bytes-per-cycle",
+        nargs="?",
+        help="""The write bandwidth of the memory pool in bytes/cycle.
+                Example usage: --workspace-pools-write-bandwidth-bytes-per-cycle=flash:8""",
+    )
+    parser.add_argument(
+        "--workspace-pools-read-latency-cycles",
+        nargs="?",
+        help="""The read latency of the memory pool in cycles.
+                Example usage: --workspace-pools-read-latency-cycles=flash:4""",
+    )
+    parser.add_argument(
+        "--workspace-pools-write-latency-cycles",
+        nargs="?",
+        help="""The write latency of the memory pool in cycles.
+                Example usage: --workspace-pools-write-latency-cycles=flash:8""",
+    )
+    parser.add_argument(
+        "--workspace-pools-target-burst-bytes",
+        help="""The burst length of the memory pool in bytes per target.
+                Example usage: --workspace-pools-target-burst-bytes=flash:accel:1""",
+    )
+
+
+def _parse_target_burst(attr, pool_name):
+    if pool_name not in attr:
+        return {}
+
+    return {target: int(attr[pool_name][target]) for target in attr[pool_name]}
+
+
+def _parse_target_access(attr, pool_name):
+    try:
+        return attr[pool_name]
+    except:
+        raise TVMCException(f'Workspace Pool "{pool_name}" using undefined Target.')
+
+
+def _split_pools(attr):
+    return re.split(",", attr) if attr else []
+
+
+def _target_attributes_to_pools(attr, targets):
+    if not targets or attr is None:
+        return {}
+
+    target_attributes = {}
+    for pool_values in re.split(",", attr):
+        pool_name, target_name, target_value = re.split(":", pool_values)
+        if pool_name not in target_attributes:
+            target_attributes[pool_name] = {}
+
+        matched_targets = [target for target in targets if target_name == target.kind.name]
+        if matched_targets:
+            target_attributes[pool_name][matched_targets[0]] = target_value
+        else:
+            raise TVMCException(
+                "The workspace pool target specification for target access "
+                "needs to be the same TVM target as when specifying targets to use."
+            )
+
+    return target_attributes
+
+
+def _attribute_to_pools(attr):
+    return dict(pool.split(":", maxsplit=1) for pool in re.split(",", attr)) if attr else {}
+
+
+def workspace_pools_recombobulate(parsed, targets):
+    """Reconstructs the Workspace Pools args and returns a WorkspaceMemoryPool object"""
+    WORKSPACE_POOL_PARAMS = [
```

Comment:
could we mark where this list comes from and include "keep in sync with" comments in both places?